src/mlpack/tests/feedforward_network_test.cpp

```diff
@@ -620,4 +620,47 @@ BOOST_AUTO_TEST_CASE(CustomLayerTest)
   arma::mat prediction = arma::zeros<arma::mat>(1, predictionTemp.n_cols);
 }
 
+/**
+ * Test the overload of Forward function which allows partial forward pass.
+ */
+BOOST_AUTO_TEST_CASE(PartialForwardTest)
+{
+  FFN<NegativeLogLikelihood<>, RandomInitialization> model;
+  model.Add<Linear<> >(5, 10);
+
+  // Add a new Add<> module which adds a constant term to the input.
+  Add<>* module = new Add<>(10);
+  model.Add(module);
+
+  model.Add<IdentityLayer<> >();
+  model.Add<Linear<> >(10, 10);
+
+  model.ResetParameters();
+  // Set the parameters of the Add<> module to a matrix of ones.
+  module->Parameters() = arma::ones(10, 1);
+
+  arma::mat input = arma::ones(5, 1);
+  arma::mat output;
+
+  // Forward pass only through the Add module.
+  model.Forward(input,
+                output,
+                1 /* Index of the Add module */,
+                1 /* Index of the Add module */);
+
+  // As we only forward pass through Add module, input and output should
+  // differ by a matrix of ones.
+  CheckMatrices(input, output - 1);
+
+  // Forward pass only through the Add module and the Identity layer.
+  model.Forward(input,
+                output,
+                1 /* Index of the Add module */,
+                2 /* Index of the Identity layer */);
```

Comment:
Do you mind to add another test, that includes the linear layer as well into the output, we can set the weights to ones as well; what I'd like to see is that the output changed. Let's suppose the end index doesn't work as expected the output is still the same as the first one since the second layer is an Identity layer.